packages/select-react/src/Select.tsx

```diff
@@ -70,19 +72,19 @@ export function Select({
     width,
     ...selectProps
 }: Props) {
-    const [selectedValue, setSelectedValue] = useState(value);
-    const [internalFocus, setInternalFocus] = useState(false);
-    const hasSelectedValue = typeof selectedValue !== "undefined" && selectedValue !== "";
+    const [searchValue, setSearchValue] = useState("");
+    const hasSelectedValue = typeof value !== "undefined" && value !== "";
 
-    const getLabelFromValue = useCallback(
-        (value: string | undefined) => {
-            const matchingItem = items.map(getValuePair).filter((item) => item.value === value)[0];
-            return matchingItem && matchingItem.label;
-        },
-        [items],
-    );
-    const [displayedValue, setDisplayedValue] = useState(getLabelFromValue(value));
+    const itemsWithVisibility = items.map(getValuePair).map((item) => {
+        const visible =
+            !searchAble || searchValue === "" || item.label.toLowerCase().indexOf(searchValue.toLowerCase()) > -1;
+        return { ...item, visible };
+    });
+    const selectedValueLabel = itemsWithVisibility.find((item) => item.value == value)?.label || defaultPrompt;
```

Comment:
```suggestion
    const selectedValueLabel = itemsWithVisibility.find((item) => item.value === value)?.label || defaultPrompt;
```

hvis intensjonen faktisk er dobbel lik, så vil jeg gjerne ha en liten kommentar